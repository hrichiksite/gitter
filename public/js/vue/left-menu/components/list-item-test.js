const mount = require('../../__test__/vuex-mount');
const { default: ListItem } = require('./list-item.vue');

const {
  createSerializedRoomFixture,
  createSerializedOneToOneRoomFixture
} = require('../../__test__/fixture-helpers');

describe('list-item', () => {
  it('community room matches snapshot', () => {
    const { wrapper } = mount(ListItem, {
      item: createSerializedRoomFixture('my-community/community')
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('room with short name matches snapshot', () => {
    const { wrapper } = mount(ListItem, {
      item: createSerializedRoomFixture('foo/bar')
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('room with long name matches snapshot', () => {
    const { wrapper } = mount(ListItem, {
      item: createSerializedRoomFixture('abcdefghijklmnop/qrstuvwxyz')
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('one to one room matches snapshot', () => {
    const { wrapper } = mount(ListItem, {
      item: createSerializedOneToOneRoomFixture('EricGitterTester')
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('favourite room matches snapshot', () => {
    const { wrapper } = mount(ListItem, {
      item: {
        ...createSerializedRoomFixture('my-community/community'),
        favourite: 1
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('active room matches snapshot', () => {
    const { wrapper } = mount(ListItem, {
      item: createSerializedRoomFixture('my-community/room1'),
      active: true
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('room with unread matches snapshot', () => {
    const { wrapper } = mount(ListItem, {
      item: {
        ...createSerializedRoomFixture('my-community/community'),
        unreadItems: 7
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('room with mentions matches snapshot', () => {
    const { wrapper } = mount(ListItem, {
      item: {
        ...createSerializedRoomFixture('my-community/community'),
        mentions: 7
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('muted room with activity matches snapshot', () => {
    const { wrapper } = mount(ListItem, {
      item: {
        ...createSerializedRoomFixture('my-community/community'),
        lurk: true,
        activity: true
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('favourite loading room matches snapshot', () => {
    const { wrapper } = mount(ListItem, {
      item: {
        ...createSerializedRoomFixture('my-community/community'),
        favourite: 1,
        loading: true
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('calls store action "changeDisplayedRoomById" and "toggleLeftMenu" when item is clicked', () => {
    const room = createSerializedRoomFixture('my-community/community');
    const { wrapper, stubbedActions } = mount(ListItem, {
      item: room
    });

    wrapper.find({ ref: 'link' }).trigger('click');

    expect(stubbedActions.changeDisplayedRoomById).toHaveBeenCalledWith(
      expect.anything(),
      room.id,
      undefined
    );

    expect(stubbedActions.toggleLeftMenu).toHaveBeenCalledWith(expect.anything(), false, undefined);
  });
});
