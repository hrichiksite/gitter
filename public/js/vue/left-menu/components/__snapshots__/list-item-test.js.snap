// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list-item active room matches snapshot 1`] = `
<li
  class="list-item"
  data-id="5a8739841543b98772a686a9-my-community/room1"
>
  <a
    class="list-item-link is-active"
    href="/my-community/room1"
  >
    <img
      class="list-item-avatar"
      src="http://localhost:5000/api/private/avatars/group/i/5a8739841543b98772a686a8"
    />
     
    <h2
      class="list-item-name"
    >
      <span
        class="name-piece"
      >
        room1
      </span>
    </h2>
     
    <!---->
     
    <!---->
  </a>
</li>
`;

exports[`list-item community room matches snapshot 1`] = `
<li
  class="list-item"
  data-id="5a8739841543b98772a686a9-my-community/community"
>
  <a
    class="list-item-link"
    href="/my-community/community"
  >
    <img
      class="list-item-avatar"
      src="http://localhost:5000/api/private/avatars/group/i/5a8739841543b98772a686a8"
    />
     
    <h2
      class="list-item-name"
    >
      <span
        class="name-piece"
      >
        my-community
      </span>
    </h2>
     
    <!---->
     
    <!---->
  </a>
</li>
`;

exports[`list-item favourite loading room matches snapshot 1`] = `
<li
  class="list-item is-favourited"
  data-favourite="1"
  data-id="5a8739841543b98772a686a9-my-community/community"
>
  <a
    class="list-item-link"
    href="/my-community/community"
  >
    <img
      class="list-item-avatar"
      src="http://localhost:5000/api/private/avatars/group/i/5a8739841543b98772a686a8"
    />
     
    <h2
      class="list-item-name"
    >
      <span
        class="name-piece"
      >
        my-community
      </span>
    </h2>
     
    <loading-spinner-stub />
     
    <!---->
  </a>
</li>
`;

exports[`list-item favourite room matches snapshot 1`] = `
<li
  class="list-item is-favourited"
  data-favourite="1"
  data-id="5a8739841543b98772a686a9-my-community/community"
>
  <a
    class="list-item-link"
    href="/my-community/community"
  >
    <img
      class="list-item-avatar"
      src="http://localhost:5000/api/private/avatars/group/i/5a8739841543b98772a686a8"
    />
     
    <h2
      class="list-item-name"
    >
      <span
        class="name-piece"
      >
        my-community
      </span>
    </h2>
     
    <!---->
     
    <!---->
  </a>
</li>
`;

exports[`list-item muted room with activity matches snapshot 1`] = `
<li
  class="list-item"
  data-id="5a8739841543b98772a686a9-my-community/community"
>
  <a
    class="list-item-link"
    href="/my-community/community"
  >
    <img
      class="list-item-avatar"
      src="http://localhost:5000/api/private/avatars/group/i/5a8739841543b98772a686a8"
    />
     
    <h2
      class="list-item-name"
    >
      <span
        class="name-piece"
      >
        my-community
      </span>
    </h2>
     
    <!---->
     
    <div
      class="activity-indicator"
    />
  </a>
</li>
`;

exports[`list-item one to one room matches snapshot 1`] = `
<li
  class="list-item"
  data-id="5c17fc82b799f1de19e23f15-EricGitterTester"
>
  <a
    class="list-item-link"
    href="/EricGitterTester"
  >
    <img
      class="list-item-avatar"
      src="http://localhost:5000/api/private/avatars/gh/uv/4/EricGitterTester"
    />
     
    <h2
      class="list-item-name"
    >
      <span
        class="name-piece"
      >
        EricGitterTester
      </span>
    </h2>
     
    <!---->
     
    <!---->
  </a>
</li>
`;

exports[`list-item room with long name matches snapshot 1`] = `
<li
  class="list-item"
  data-id="5a8739841543b98772a686a9-abcdefghijklmnop/qrstuvwxyz"
>
  <a
    class="list-item-link"
    href="/abcdefghijklmnop/qrstuvwxyz"
  >
    <img
      class="list-item-avatar"
      src="http://localhost:5000/api/private/avatars/group/i/5a8739841543b98772a686a8"
    />
     
    <h2
      class="list-item-name"
    >
      <span
        class="name-piece"
      >
        qrstuvwxyz
      </span>
    </h2>
     
    <!---->
     
    <!---->
  </a>
</li>
`;

exports[`list-item room with mentions matches snapshot 1`] = `
<li
  class="list-item"
  data-id="5a8739841543b98772a686a9-my-community/community"
>
  <a
    class="list-item-link"
    href="/my-community/community"
  >
    <img
      class="list-item-avatar"
      src="http://localhost:5000/api/private/avatars/group/i/5a8739841543b98772a686a8"
    />
     
    <h2
      class="list-item-name"
    >
      <span
        class="name-piece"
      >
        my-community
      </span>
    </h2>
     
    <!---->
     
    <div
      class="mention-indicator"
    >
      @
    </div>
  </a>
</li>
`;

exports[`list-item room with short name matches snapshot 1`] = `
<li
  class="list-item"
  data-id="5a8739841543b98772a686a9-foo/bar"
>
  <a
    class="list-item-link"
    href="/foo/bar"
  >
    <img
      class="list-item-avatar"
      src="http://localhost:5000/api/private/avatars/group/i/5a8739841543b98772a686a8"
    />
     
    <h2
      class="list-item-name"
    >
      <span
        class="name-piece"
      >
        foo/bar
      </span>
    </h2>
     
    <!---->
     
    <!---->
  </a>
</li>
`;

exports[`list-item room with unread matches snapshot 1`] = `
<li
  class="list-item"
  data-id="5a8739841543b98772a686a9-my-community/community"
>
  <a
    class="list-item-link"
    href="/my-community/community"
  >
    <img
      class="list-item-avatar"
      src="http://localhost:5000/api/private/avatars/group/i/5a8739841543b98772a686a8"
    />
     
    <h2
      class="list-item-name"
    >
      <span
        class="name-piece"
      >
        my-community
      </span>
    </h2>
     
    <!---->
     
    <div
      class="unread-indicator"
    >
      
      7
    
    </div>
  </a>
</li>
`;
